### ISSUE TITLE
order exports

### ISSUE THREAD

**Issue by gajus** [benmosher/eslint-plugin-import#1888]
Surprisingly do not see an existing issue for this, but I would like to have an equivalent of `order` with `alphabetize` for exports.

This would esp. useful for files that only contain exports, e.g.

```js
// @flow

export {
  default as createLogDump,
} from './createLogDump';
export {
  default as createSnackTimeEvent,
} from './createSnackTimeEvent';
export {
  default as executeMaintenanceTask,
} from './executeMaintenanceTask';
export {
  default as inviteSlackTeamUsersToSnackClubChannel,
} from './inviteSlackTeamUsersToSnackClubChannel';
export {
  default as pairSnackTimeEventParticipants,
} from './pairSnackTimeEventParticipants';

```

At the moment, these exports are in a random order.

Enforcing export order when file only contains exports would be useful.

**Comment by tfoxy** [comment]
Related to #659. In that case it's for exported members, but some people are commenting libraries that solve the exported module order issue.

**Comment by aaronadamsCA** [comment]
@gajus Is it fair to say your request here is limited to re-exports, aka `export ... from`?

That's all I care about personally. Given re-export syntax appears to be a subset of import syntax, it might be a far simpler request to fulfill.

**Comment by gajus** [comment]
Correct. If anyone is keen to contribute, consider adding it to https://github.com/gajus/eslint-plugin-canonical

**Comment by gajus** [comment]
or that. I only suggest eslint-plugin-canonical because it is a playground to quickly release and test rules. `eslint-plugin-import` is more mature and careful in its release of features.

**Comment by guillaumeduboc** [comment]
Hey all,
I'd love to see this in `eslint-plugin-import` and I started looking into it

I listed all these kind of re-exports I'd like to handle (I may have forgotten some) 

```typescript
export * from "module-name";
export * as name1 from "module-name";

export { default } from "module-name";

export { name1, /* …, */ nameN } from "module-name";
export { import1 as name1, import2 as name2, /* …, */ nameN } from "module-name";
export { default as name1 } from "module-name";

export type * from "module-name";

export type { name1 } from "module-name";
```

@ljharb do you imagine the exports should be grouped (same as imports) ?
for the moment I'm thinking about creating a new rule with only the order but eventually adding grouping, new lines, etc...

**Comment by ljharb** [comment]
I would assume that all the same options for imports would govern exports, whenever it makes sense.

**Comment by guillaumeduboc** [comment]
I suggest we do a different rule, but I might move some logic around
I was thinking putting some of the `order` logic in `core`

**Comment by carlocorradini** [comment]
Super! Any ETA? 🥳

**Comment by guillaumeduboc** [comment]
no idea yet, I've just started looking into it 😁
